src/client/app/components/BarChartComponent.tsx

```diff
@@ -0,0 +1,257 @@
+					const xData: string[] = [];
+					const yData: number[] = [];
+					const hoverText: string[] = [];
+					const readings = _.values(readingsData);
+					readings.forEach(barReading => {
+						const st = moment.utc(barReading.startTimestamp);
+						// Time reading is in the middle of the start and end timestamp (may change this depending on how it looks on the bar graph)\
+						const timeReading = st.add(moment.utc(barReading.endTimestamp).diff(st) / 2);
+						xData.push(timeReading.utc().format('YYYY-MM-DD HH:mm:ss'));
+						let readingValue = barReading.reading;
+						if (selectedAreaNormalization) {
+							readingValue /= groupArea;
+						}
+						yData.push(readingValue);
+						// only display a range of dates for the hover text if there is more than one day in the range
+						let timeRange: string = `${moment.utc(barReading.startTimestamp).format('ll')}`;
+						if (barDuration.asDays() != 1) {
+							// subtracting one extra day caused by day ending at midnight of the next day.
+							// Going from DB unit timestamp that is UTC so force UTC with moment, as usual.
+							timeRange += ` - ${moment.utc(barReading.endTimestamp).subtract(1, 'days').format('ll')}`;
+						}
+						hoverText.push(`<b> ${timeRange} </b> <br> ${label}: ${readingValue.toPrecision(6)} ${unitLabel}`);
+					});
+
+					// This variable contains all the elements (x and y values, bar chart, etc.) assigned to the data parameter of the Plotly object
+					datasets.push({
+						name: label,
+						x: xData,
+						y: yData,
+						text: hoverText,
+						hoverinfo: 'text',
+						type: 'bar',
+						marker: { color: getGraphColor(colorID, DataType.Group) }
+					});
+				}
+			}
+		}
+	}
+
+	const handleRelayout = (e: PlotRelayoutEvent) => {
+		// This event emits an object that contains values indicating changes in the user's graph, such as zooming.
+		// These values indicate when the user has zoomed or made other changes to the graph.
+		if (e['xaxis.range[0]'] && e['xaxis.range[0]']) {
+			// The event signals changes in the user's interaction with the graph.
+			// this will automatically trigger a refetch due to updating a query arg.
+			const startTS = moment.utc(e['xaxis.range[0]'])
+			const endTS = moment.utc(e['xaxis.range[1]'])
+			const workingTimeInterval = new TimeInterval(startTS, endTS);
+			dispatch(graphSlice.actions.updateTimeInterval(workingTimeInterval));
+		}
+	}
+
+	// Assign all the parameters required to create the Plotly object (data, layout, config) to the variable props, returned by mapStateToProps
+	// The Plotly toolbar is displayed if displayModeBar is set to true (not for bar charts)
+
+	if (raw) {
+		return <h1><b>${translate('bar.raw')}</b></h1>
+	}
+	let enoughData = false;
+	datasets.forEach(dataset => {
+		if (dataset.x.length > 1) {
+			enoughData = true
+		}
+	})
+	if (datasets.length === 0) {
+		return <h1>
+			{`${translate('select.meter.group')}`}
+		</h1>
+	} else if (!enoughData) {
+		// This normal so plot.
```

Comment:
First, this does not seem a normal plot as it has no data. Second, I think we may want to change the translate key since not just 3D. Let me know and I can do if you want.

Reply:
Yes please fix the translations to whatever you find suitable. Also, yes the comment is in the wrong location.

Reply:
done